driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -74,7 +74,12 @@ public void connect() throws UnsupportedProtocolVersionException {
             return;
 
         // NB: at this stage, allHosts() only contains the initial contact points
-        setNewConnection(reconnectInternal(cluster.metadata.allHosts().iterator(), true));
+        List<Host> hosts = new ArrayList<Host>(cluster.metadata.allHosts());
+        // shuffle contact points to spread load evenly across hosts on the very first connection;
```

Comment:
"spread load" is maybe a bit deceptive because the driver is not under any kind of load at init, I suggest rewording to "shuffle so that multiple clients with the same contact points don't all pick the same control host".